compat/mingw.c

```diff
@@ -3021,6 +3043,18 @@ static void setup_windows_environment(void)
 	if (!(tmp = getenv("MSYS")) || !strstr(tmp, "winsymlinks:nativestrict"))
 		has_symlinks = 0;
 
+	/* Check if this release of Windows supports creating symbolic links as an
+	 * unprivileged user. The underlying API support was introduced in the
+	 * Windows 10 Creators Update (v1703). Older releases of Windows don't
+	 * understand the new flag to the CreateSymbolicLink family of functions.
+	 */
+	struct utsname winver;
+	if (uname(&winver) == 0) {
+		if (atoi(winver.version) >= 15063) {
+			has_unprivileged_symlinks = 1;
```

Comment:
That's true, but to support the new functionality we only need to check the OS version supports the API improvements to the `CreateSymbolicLink` family of functions so we can pass in the additional flag. If Developer Mode is not enabled, passing the flag won't have any effect, so there's no behaviour change from the current arrangement.